examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,680 @@
+    executorService.schedule(() -> {
+        discoverer.stopDiscovery();
+        discoverer.removeCastingPlayerChangeListener(castingPlayerChangeListener);
+    },     30, TimeUnit.SECONDS);
+}
+```
+
+On iOS, register the listeners by calling `addObserver` on the
+`NotificationCenter` with the appropriate selector, and then call `start` on the
+`sharedInstance` of `MTRCastingPlayerDiscovery`.
+
+```objectivec
+func startDiscovery() {
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.addDiscoveredCastingPlayers),
+    name: NSNotification.Name.didAddCastingPlayers,
+    object: nil)
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.removeDiscoveredCastingPlayers),
+    name: NSNotification.Name.didRemoveCastingPlayers,
+    object: nil)
+    NotificationCenter.default.addObserver(
+    self,
+    selector: #selector(self.updateDiscoveredCastingPlayers),
+    name: NSNotification.Name.didUpdateCastingPlayers,
+    object: nil)
+
+    MTRCastingPlayerDiscovery.sharedInstance().start()
+    self.discoveryRequestStatus = true
+}
+```
+
+Note: You will need to connect with a Casting Player as described below to see
+the list of Endpoints that they support. Refer to the [Connection](#connection)
+section for details on how to discover available endpoints supported by a
+Casting Player.
+
+### Connect to a Casting Player
+
+_{Complete Connection examples:
+[Android](android/App/app/src/main/java/com/matter/casting/ConnectionExample.java)
+| [iOS](darwin/TvCasting/TvCasting/MTRConnectionExampleViewModel.swift)}_
+
+Each CastingPlayer object created during [Discovery](#discovery) contains
+information such as deviceName, vendorId, productId, etc. which can help the
+user pick the right CastingPlayer. A Casting Client can attempt to connect to
+the selectedCastingPlayer using
+[Matter User Directed Commissioning (UDC)](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/rendezvous/UserDirectedCommissioning.adoc).
+The Matter TV Casting library locally caches information required to reconnect
+to a CastingPlayer, once it has commissioned with it. After that, it is able to
+skip the full UDC process by establishing CASE with the CastingPlayer directly.
+Once connected, the CastingPlayer object would contain the list of available
+Endpoints on it.
+
+On Android, the Casting Client can connect to a `CastingPlayer` by calling
+`connect` on it, and handling any exceptions that may arise in the process.
+
+```java
+// Maximum time in seconds we'll wait for the connection to the CastingPlayer to go through
+private static final long CONNECTION_TIMEOUT_SECS = 45;
+
+public static void demoConnection(CastingPlayer castingPlayer) {
+    // Connect to the castingPlayer
+    CompletableFuture<Void> connectionFuture = castingPlayer.connect(CONNECTION_TIMEOUT_SECS);
+
+    // handle exception, if any
+    connectionFuture.exceptionally(exc -> {
```

Comment:
You may also want to add the success callback definition for the future here for demonstration purpose.

Reply:
Done